secure_message/common/utilities.py

```diff
@@ -113,11 +128,15 @@ def add_from_details(messages):
     for user in party.get_users_details(get_external_user_uuid_list(messages)):
         external_user_details[user['id']] = user
     for message in messages:
-        if message["from_internal"]:
-            message.update({"@msg_from": internal_user_service.get_user_details(message["msg_from"])})
-        else:
-            if external_user_details.get(message['msg_from']):
-                message.update({'@msg_from': external_user_details.get(message['msg_from'])})
+        try:
+            if message["from_internal"]:
+                message.update({"@msg_from": internal_user_service.get_user_details(message["msg_from"])})
+            else:
+                if external_user_details.get(message['msg_from']):
+                    message.update({'@msg_from': external_user_details.get(message['msg_from'])})
+        except IndexError:
+            logger.exception(f"Exception adding from details message:'{json.dumps(message)}'")
```

Comment:
Here too

Reply:
Also also, we should be using fields for this information.  So this would become:
```python
logger.exception("Exception adding from details message", message=json.dumps(message))
```
But message is too broad, but you get the idea ^^